tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/rules/aux/IntOracle.scala

```diff
@@ -33,21 +34,25 @@ class IntOracle(val intCell: ArenaCell, nvalues: Int) extends Oracle {
   /**
    * Produce a ground expression that contains assertions for the possible oracle values.
    *
-   * @param state      a symbolic state
-   * @param assertions a sequence of assertions, one per oracle value
+   * @param state          a symbolic state
+   * @param assertions     a sequence of assertions, one per oracle value
+   * @param elseAssertions an optional sequence of assertions, one per oracle value
    * @return an expression ite(oracle = 0, ite(oracle = 1, ...))
    */
-  override def caseAssertions(state: SymbState, assertions: Seq[TlaEx]): TlaEx = {
+  override def caseAssertions(state: SymbState, assertions: Seq[TlaEx], elseAssertions: Seq[TlaEx] = Seq()): TlaEx = {
+    if (elseAssertions.nonEmpty & assertions.size != elseAssertions.size) {
+      throw new IllegalStateException(s"Invailid call to Oracle, malformed elseAssertions")
+    }
+
     nvalues match {
       case 0 => state.arena.cellTrue().toNameEx
 
       case 1 => assertions.head
 
       case _ =>
-        val es = assertions
-          .slice(0, nvalues)
-          .zipWithIndex
-          .map { case (e, i) => tla.or(e, tla.not(whenEqualTo(state, i))) }
+        val iteCases = assertions.zipAll(elseAssertions, ValEx(TlaBool(true)), ValEx(TlaBool(true)))
```

Comment:
We could use `zip` instead of `zipAll`, since there is an assertion above.

Reply:
Can we? If `elseAssertions` is empty we have to use `zipAll`, no?

Reply:
I don't think it is necessary. If `elseAssertions` is not empty `zipAll` does nothing, if it is empty it makes the second element of each tuple in `iteCases` be `ValEx(TlaBool(true))`. If you think it would be good to explicitly populate `elseAssertions` with `assertions.size` copies of `ValEx(TlaBool(true))` an then use `zip` to make clearer what is happening I could do it.

Reply:
Maybe adding a comment would be enough.